python_modules/dagster/dagster/core/instance/__init__.py

```diff
@@ -736,19 +736,21 @@ def create_run_for_pipeline(
                     solids_to_execute=solids_to_execute
                 )
 
+        resolved_run_config = ResolvedRunConfig.build(pipeline_def, run_config, mode)
```

Comment:
This seems a little worrisome, but it does seem that the only callers are in user processes?

Reply:
Seemed ok given that other methods on this class were creating execution plan (which implicitly retrieves pipeline def). But agreed that it is kinda worrisome that this is necessary / not gated more concretely

Reply:
This can move back now